notebook/auth/security.py

```diff
@@ -118,6 +118,7 @@ def persist_config(config_file=None, mode=0o600):
     """
 
     if config_file is None:
+        os.makedirs(jupyter_config_dir(), exist_ok=True)
```

Comment:
I think this should be outside the `if` block, and use `os.path.dirname()`, so it works the same way if a path is passed in explicitly.

Reply:
fixed